### PR TITLE
feat: sheet title as template

### PR DESCRIPTION
**Pull request by itssharmasandeep**
## Description
Sheet title as a template

### Testing
Local testing is done,

### Checklist:
- [x] My changes generate no new warnings

### CODE REVIEW

File: projects/components/src/overlay/sheet/sheet-overlay.component.ts [hunk 1]
```diff
@@ -33,6 +35,9 @@ import { SheetOverlayConfig, SheetSize } from './sheet';
               >
               </ht-button>
             </div>
+            <ng-template #defaultSheetTitleTemplate>
+              <h3 class="default-header-title">{{ sheetTitle }}</h3>
```
Comment (arjunlalb) [inline projects/components/src/overlay/sheet/sheet-overlay.component.ts:39]:
```suggestion
              <h3 class="default-header-title">{{ this.sheetTitle }}</h3>
```

Comment (arjunlalb) [inline projects/components/src/overlay/sheet/sheet-overlay.component.ts:39]:
```suggestion
              <h3 class="default-header-title">{{ this.sheetTitle }}</h3>
```

File: projects/components/src/overlay/sheet/sheet-overlay.component.ts [hunk 2]
```diff
@@ -81,6 +87,7 @@ export class SheetOverlayComponent {
     this.hasDefaultPadding = sheetConfig.hasDefaultPadding ?? true;
     this.showHeader = sheetConfig.showHeader === true;
     this.sheetTitle = sheetConfig.title === undefined ? '' : sheetConfig.title;
+    this.isSheetTitleAString = typeof this.sheetTitle === 'string';
```
Comment (arjunlalb) [inline projects/components/src/overlay/sheet/sheet-overlay.component.ts:90]: would be cleaner to check the inverse with `instanceof TemplateRef`

Comment (itssharmasandeep) [reply projects/components/src/overlay/sheet/sheet-overlay.component.ts:90]: Not sure, I did the same thing at many places, so continued with the same :)

**Review by arjunlalb — APPROVED** [review]

File: projects/components/src/overlay/sheet/sheet-overlay.component.ts [hunk 3]
```diff
@@ -58,7 +63,7 @@ import { SheetOverlayConfig, SheetSize } from './sheet';
   `
 })
 export class SheetOverlayComponent {
-  public readonly sheetTitle: string;
+  public readonly sheetTitle: string | TemplateRef<unknown>;
```
Comment (anandtiwary) [inline projects/components/src/overlay/sheet/sheet-overlay.component.ts:66]: Title is supposed to be text only. If we allow it to be a template then technically we can pass any custom content here. I would propose that we keep title as it is and introduce a new property `customHeaderTemplate` that will be rendered between the title text and the header controls

File: projects/components/src/overlay/sheet/sheet-overlay.component.ts (hunk 2 above)
Comment (anandtiwary) [reply projects/components/src/overlay/sheet/sheet-overlay.component.ts:90]: we can avoid this :)

**Pull request merged**